specification/common-types/resource-management/v5/networksecurityperimeter.json

```diff
@@ -0,0 +1,400 @@
+        }
+      }
+    },
+    "ProvisioningIssue": {
+      "description": "Describes provisioning issue for a network security perimeter configuration",
+      "type": "object",
+      "readOnly": true,
+      "properties": {
+        "name": {
+          "description": "Name of the issue",
+          "type": "string",
+          "readOnly": true
+        },
+        "properties": {
+          "$ref": "#/definitions/ProvisioningIssueProperties"
+        }
+      }
+    },
+    "ProvisioningIssueProperties": {
+      "description": "Properties of a provisioning issue",
+      "type": "object",
+      "readOnly": true,
+      "properties": {
+        "issueType": {
+          "description": "Type of issue",
+          "type": "string",
+          "readOnly": true,
+          "enum": [
+            "Unknown",
+            "ConfigurationPropagationFailure"
+          ],
+          "x-ms-enum": {
+            "name": "IssueType",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "Unknown",
+                "description": "Unknown issue type"
+              },
+              {
+                "value": "ConfigurationPropagationFailure",
+                "description": "An error occurred while applying the configuration."
+              }
+            ]
+          }
+        },
+        "severity": {
+          "description": "Severity of the issue.",
+          "enum": [
+            "Warning",
+            "Error"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "Severity",
+            "modelAsString": true
+          }
+        },
+        "description": {
+          "description": "Description of the issue",
+          "type": "string",
+          "readOnly": true
+        },
+        "suggestedResourceIds": {
+          "description": "Fully qualified resource IDs of resources that can be associated to the same perimeter to remediate the issue.",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "description": "Fully qualified resource ID of the suggested resource",
+            "type": "string",
+            "format": "arm-id",
+            "readOnly": true
+          }
+        },
+        "suggestedAccessRules": {
+          "description": "Access rules that can be added to the same profile to remediate the issue.",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/AccessRule"
+          },
+          "x-ms-identifiers": []
+        }
+      }
+    },
+    "NetworkSecurityPerimeter": {
+      "description": "Information about a network security perimeter (NSP)",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "Fully qualified Azure resource ID of the NSP resource",
+          "type": "string",
+          "format": "arm-id"
```

Comment:
Is there an allowed resource type restriction we can add to this?

Reply:
I am imagining its supposed to be a `Microsoft.Network/networkSecurityPerimeters` but I don't really know if there are any other possibilities.

Reply:
Technically, adding the constraint later will be a breaking change, while opening it up will be non-breaking. Feels like it's worth it to add it up front.